packages/@aws-cdk/aws-apigatewayv2/lib/api.ts

```diff
@@ -0,0 +1,186 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { CfnApi, CfnApiProps,  HttpRouteOptions, IRouteBase, LambdaRouteOptions, Route  } from '../lib';
+
+/**
+ * the HTTP API interface
+ */
+export interface IHttpApi extends cdk.IResource {
+  /**
+   * The ID of this API Gateway HTTP Api.
+   * @attribute
+   */
+  readonly httpApiId: string;
+}
+
+abstract class ApiBase extends cdk.Resource implements IHttpApi {
+  /**
+   * API ID
+   */
+  public abstract readonly httpApiId: string;
+  /**
+   * API URL
+   */
+  public abstract readonly url: string;
+}
+
+/**
+ * API properties
+ */
+export interface HttpApiProps extends cdk.StackProps {
+  /**
+   * A name for the HTTP API resoruce
+   *
+   * @default - ID of the HttpApi construct.
+   */
+  readonly apiName?: string;
```

Comment:
this is mandatory.

Reply:
If `apiName` is a mandatory property of `HttpApiProps` then when we `new HttpApi()` we always need to specify an `apiName` for it. Is it correct?

Reply:
Sorry I wasn't clear.

The [`Name` parameter in `ApiGatewayv2::Api`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-api.html#cfn-apigatewayv2-api-name) is mandatory (unless an OpenAPI definition is provided).

What is it going to be in the case that a user does not specify it? The CDK has to set some value here, correct? The `@default` annotation says the ID of the `HttpApi` construct, but that isn't available at the time of construction of `CfnApi`.

Reply:
Also, this should be named `httpApiName`.